src/kagglehub/http_resolver.py

```diff
@@ -337,12 +333,18 @@ def _build_get_notebook_url_path(h: NotebookHandle) -> str:
     return f"kernels/pull?user_name={h.owner}&kernel_slug={h.notebook}"
 
 
-def _build_dataset_download_url_path(h: DatasetHandle) -> str:
-    return f"datasets/download/{h.owner}/{h.dataset}?dataset_version_number={h.version}"
+def _build_dataset_download_url_path(h: DatasetHandle, path: Optional[str]) -> str:
+    if path:
+        return f"datasets/download/{h.owner}/{h.dataset}?dataset_version_number={h.version}&file_name={path}"
+    else:
+        return f"datasets/download/{h.owner}/{h.dataset}?dataset_version_number={h.version}"
 
 
-def _build_notebook_download_url_path(h: NotebookHandle) -> str:
-    return f"kernels/output/download/{h.owner}/{h.notebook}?version_number={h.version}"
+def _build_notebook_download_url_path(h: NotebookHandle, path: Optional[str]) -> str:
+    if path:
+        return f"kernels/output/download/{h.owner}/{h.notebook}?version_number={h.version}&file_path={path}"
+    else:
+        return f"kernels/output/download/{h.owner}/{h.notebook}?version_number={h.version}"
```

Comment:
nit: Minor, but for all three functions, I'd probably lean towards following the behavior used elsewhere in this repo of starting with the `base_url = ...` then `if path: return f"{base_url}/{path}"` case, to avoid duplication.